python/lsst/daf/butler/core/dimensions/_config.py

```diff
@@ -62,11 +63,81 @@ class DimensionConfig(ConfigSubset):
 
     See the documentation for the linked classes above for more information
     on the configuration syntax.
+
+    Parameters
+    ----------
+    other : `Config` or `str` or `dict`, optional
+        Argument specifying the configuration information as understood
+        by `Config`. If `None` is passed then defaults are loaded from
+        "dimensions.yaml", otherwise defaults are not loaded.
+    validate : `bool`, optional
+        If `True` required keys will be checked to ensure configuration
+        consistency.
+    searchPaths : `list` or `tuple`, optional
+        Explicit additional paths to search for defaults. They should
+        be supplied in priority order. These paths have higher priority
+        than those read from the environment in
+        `ConfigSubset.defaultSearchPaths()`.  Paths can be `str` referring to
+        the local file system or URIs, `ButlerURI`.
     """
-    component = "dimensions"
+
     requiredKeys = ("version", "elements", "skypix")
     defaultConfigFile = "dimensions.yaml"
 
+    def __init__(self, other: Union[Config, ButlerURI, str, None] = None, validate: bool = True,
+                 searchPaths: Optional[Iterable[Union[str, ButlerURI]]] = None):
+        # if argument is not None then do not load/merge defaults
+        mergeDefaults = other is None
+        super().__init__(other=other, validate=validate, mergeDefaults=mergeDefaults,
+                         searchPaths=searchPaths)
+
+    def _updateWithConfigsFromPath(self, searchPaths: Iterable[Union[str, ButlerURI]],
```

Comment:
Rather than copying all this code, can we have a class property that DimensionConfig can set to indicate whether we are finding first or merging all found?

Reply:
I thought about but did not want to mess with the base class just yet. But you are right, it's easier to do via policy in the base class implementation.